core/src/test/resources/sql/cast.iq

```diff
@@ -1918,4 +1918,43 @@ EnumerableCalc(expr#0=[{inputs}], expr#1=[1.11:DECIMAL(3, 2)], expr#2=[2.22:DECI
   EnumerableValues(tuples=[[{ 0 }]])
 !plan
 
+# [CALCITE-7062]
+# Common type for nullable date and not nullable timestamp
+SELECT CAST(NULL AS DATE) UNION ALL SELECT TIMESTAMP '2025-07-04 10:00:00';
++---------------------+
+| EXPR$0              |
++---------------------+
+|                     |
+| 2025-07-04 10:00:00 |
++---------------------+
+(2 rows)
+
+!ok
+
+# [CALCITE-7062]
+# Common type for null and not nullable array;
+SELECT ARRAY[1, 2, 3] UNION ALL SELECT NULL;
```

Comment:
We need to add additional test cases such as:
```
SELECT ARRAY[cast(null as integer), cast(null as integer), cast(null as integer)] UNION ALL SELECT NULL;
```
Or:
```
SELECT ARRAY[cast(null as integer), cast(null as integer), cast(null as integer)] UNION ALL SELECT ARRAY[1, 2, 3];
```

Reply:
Test added